src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -3756,6 +3762,35 @@ def should_enable_distributed_tracing(consumption_plan_location, matched_runtime
         and image is None
 
 
+def update_functionapp_polling(cmd, resource_group_name, name, functionapp):
+    try:
+        _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'update', None, functionapp)
+    except Exception as ex:  # pylint: disable=broad-except
+        poll_url = ex.response.headers['Location'] if 'Location' in ex.response.headers else None
+        if ex.response.status_code == 202 and poll_url:
+            r = send_raw_request(cmd.cli_ctx, method='get', url=poll_url)
+            poll_timeout = time.time() + 60 * 2  # 2 minute timeout
+
+            while r.status_code != 200 and time.time() < poll_timeout:
+                time.sleep(5)
+                r = send_raw_request(cmd.cli_ctx, method='get', url=poll_url)
+        else:
+            raise CLIError(ex)
+
+
+def update_resource_config(cmd, resource_group_name, name, workload_profile_name=None, cpu=None, memory=None):
```

Comment:
Also, might be a good idea to rename this function since it is not clear just from the name what exactly would be updated. I would update it to something like `update_workload_profile_config`